src/ShaderParam_TEST.cc

```diff
@@ -87,6 +87,115 @@ TEST(ShaderParam, IntType)
   EXPECT_EQ(10, val3);
 }
 
+/////////////////////////////////////////////////
+TEST(ShaderParam, FloatBufferType)
+{
+  float b[10] =
+  {
+    0.0f, 0.1f, 0.2f, 0.3f, 0.4f, 0.5f, 0.6f, 0.7f, 0.8f, 0.9f
+  };
+
+  ShaderParam p;
+  p.InitializeWordBuffer(10);
+  p.UpdateWordBuffer(b);
+
+  EXPECT_EQ(ShaderParam::PARAM_FLOAT_BUFFER, p.Type());
+  EXPECT_EQ(10, (int)p.Count());
+
+  std::shared_ptr<void> wordBuffer;
+  EXPECT_TRUE(p.WordBuffer(&wordBuffer));
+  for (int x = 0; x < 10; ++x)
+  {
+    EXPECT_FLOAT_EQ((float)x / 10.0f,
```

Comment:
we like to avoid C style cast. Can you change this (and other places) to `static_cast<float>`?